### PR TITLE
Add Libretro Frontends

### PR DESCRIPTION
**Pull request by RobLoach**
A bunch of this was taken from the Wiki, and @Alcaro's list...
- https://wiki.libretro.com/index.php?title=Frontends
- https://github.com/Alcaro/misctoys/blob/master/libretrofrontlist.md

### CODE REVIEW

**Comment by Alcaro** [comment]
https://wiki.gnome.org/Apps/Games

https://github.com/mudlord/einweggerat

**Comment by RobLoach** [comment]
Good find!

**Comment by Alcaro** [comment]
oh, you had GNOME already. I somehow clicked the wiki and thought that was your PR, and listed the notable ones I know of that weren't there.

If you're gonna include noarch, half of the test fronts in minir (subproj/) count; retroprofile is noarch except runs multiple frames, retrorepeat is some kind of test for whether the core is deterministic, retrostateverify is retrorepeat on steroids (it scans the process' address space and finds exactly which variable is improperly stated).

...if they still compile, I've changed the internal APIs many times and often forget updating them. Probably broken. Probably easy to fix if I cared. Probably easy to fix if anyone else cares.

**Comment by RobLoach** [comment]
We could add them as notes in Minir if you don't think they need their own row. Seem like a good thing to mention. Can preview it over at:
https://github.com/RobLoach/docs/blob/frontends/docs/tech/frontends.md

Added you on there so feel free to edit directly.

**Comment by Alcaro** [comment]
Let's sort by purpose rather than usecase. Demo/test fronts go below the ones intended for actual use.

With mudlord conveniently in the middle. He claims it's for debugging, but it looks more like actual use to me, so this way, neither of us is wrong.

Also added minir cores. They don't need three rows, but if noarch deserves a row, so do they. As far as end users are concerned, they're not part of minir, and shouldn't be together.

File: docs/tech/frontends.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ Libretro frontends are programs that have implemented the [libretro API](../spec
 Name | Author(s) | Last Release | Description
 ------|-----------|--------------|------------
 [RetroArch](http://retroarch.com) | [libretro](http://libretro.com) | 2017-11-21 1.6.9 | Official reference libretro frontend, developed in-house. It is usually the first in implementing new features added to the libretro API. Written almost entirely in C, targets a large amount of platforms.
-[Minir](https://github.com/Alcaro/minir) | Alcaro | | WIMP interface (Windows, Icons, Menus and Pointers), and only cares about the major desktop OSes. Drops flexibility in favor of improved out-of-the-box experience.
+[minir](https://github.com/Alcaro/minir) | Alcaro | | WIMP interface (Windows, Icons, Menus and Pointers), and only cares about the major desktop OSes. Drops flexibility in favor of improved out-of-the-box experience.
```
Comment (RobLoach) [inline docs/tech/frontends.md]: thank You!

**Pull request merged**